api-client/lib/api/extensions/get_into_teaching_api_client/api_client.rb

```diff
@@ -62,20 +74,35 @@ def object_to_hash(obj)
         end
       end
 
-    private
+      private
 
       def format_date_times(params = {})
         params.transform_values do |value|
           case value
-            when DateTime, Time
-              value.strftime(API_DATE_TIME_FORMAT) 
-            when Date
-              value.strftime(API_DATE_FORMAT)
-            else
-              value
+          when DateTime, Time
+            value.strftime(API_DATE_TIME_FORMAT)
+          when Date
+            value.strftime(API_DATE_FORMAT)
+          else
+            value
           end
         end
       end
+
+      def error_breaks_circuit?(exception, handler)
+        exceptions = [Faraday::UnauthorizedError,
+                      Faraday::ForbiddenError,
+                      Faraday::ServerError]
+
+        raise exception unless exceptions.include?(exception.class)
+
+        handler.call(exception)
+      end
+
+      def raise_circuit_broken_error(exception)
+        raise exception if exception.present?
```

Comment:
This needs a comment as its not intuitive; we're re-raising Faraday exceptions while the circuit is in the process of breaking so that the client app behaviour doesn't change and `exception` being `nil` indicates that the circuit breaker middleware has hijacked the response and would otherwise return a `Faraday::Response` with a `503` HTTP status (where we want to instead return a custom `CircuitBrokenError`).